projects/ngx-translate/core/src/lib/translate.pipe.ts

```diff
@@ -11,9 +12,9 @@ export class TranslatePipe implements PipeTransform, OnDestroy {
   value: string = '';
   lastKey: string;
   lastParams: any[];
-  onTranslationChange: EventEmitter<TranslationChangeEvent>;
-  onLangChange: EventEmitter<LangChangeEvent>;
-  onDefaultLangChange: EventEmitter<DefaultLangChangeEvent>;
+  onTranslationChange: Subscription;
```

Comment:
Just curious: how comes this was changed from `EventEmitter` to `Subscription`?

Reply:
The `angular/core` team changed types for EventEmitter, relinquishing `any` and making these ones out-of-the-place:
https://github.com/angular/angular/commit/aebec4b1560063718a4d0cf1c42a8908668c0463#diff-c2e39457be0b071e3b8a08e84bc56583

Reply:
I don't quite understand this either. The `any` type was updated to the `Subscription` type for the `.subscribe()` method of the `EventEmitter`. The `EventEmiter` type itself didn't change. I think this should have stayed unchanged

Reply:
I was under impression these are assignments for results of `subscribe` calls, later unsubscribed in the `_dispose`, not references for event emitter instances created to expose change state outside. I didn't go back to trace why the event emitter was used here in the first place. I understand purpose of the `EventEmitter` when used in the translate store and service, but here I'm not sure (what's the reason for me to subscribe here to detect changes in pipe state?)
This does not compile after upgrade (when core lib changes):
```text
Type 'Subscription' is missing the following properties from type 'EventEmitter<TranslationChangeEvent>': __isAsync, emit, subscribe, observers, and 16 more.
Type 'Subscription' is missing the following properties from type 'EventEmitter<LangChangeEvent>': __isAsync, emit, subscribe, observers, and 16 more.
Type 'Subscription' is missing the following properties from type 'EventEmitter<DefaultLangChangeEvent>': __isAsync, emit, subscribe, observers, and 16 more.
```
Simply put, this stops build fo the lib. This is a pipe and I don't see what's the case of using EventEmitter to expose pipe state change.

Reply:
I can confirm that this change is necessary, you shouldn't use `EventEmitter` outside of `@Output`